workflow/rules/trimming.smk

```diff
@@ -10,25 +12,48 @@ rule get_sra:
 
 def get_raw_fastq(wildcards):
     unit = units.loc[wildcards.sample].loc[wildcards.unit]
+    if unit["fq1"].endswith("gz"):
+        ending = ".gz"
+    else:
+        ending = ""
+
     if pd.isna(unit["fq1"]):
         # SRA sample (always paired-end for now)
         accession = unit["sra"]
         return expand("sra/{accession}_{read}.fastq", accession=accession, read=[1, 2])
     if pd.isna(unit["fq2"]):
         # single end local sample
-        return unit["fq1"].values
+        return f"pipe/cutadapt/{unit.sample_name}-{unit.unit_name}.fq1.fastq{ending}"
     else:
         # paired end local sample
-        return unit[["fq1", "fq2"]].values
+        return expand(f"pipe/cutadapt/{unit.sample_name}-{unit.unit_name}.{{read}}.fastq{ending}", read=["fq1","fq2"])
+
+
+def cutadapt_pipe_input(wc):
+    files = list(sorted(glob.glob(units.loc[wc.sample].loc[wc.unit, wc.fq])))
+    #print(wc, units.loc[wc.sample].loc[wc.unit, wc.fq])
+    assert(len(files) > 0)
+    return files
+
+
+rule cutadapt_pipe:
+    threads:
+        0
+    input:
+        cutadapt_pipe_input
+    output:
+        pipe("pipe/cutadapt/{sample}-{unit}.{fq}.fast{ending}")
+    shell:
+        "cat {input} > {output}"
 
 
 rule cutadapt_pe:
     input:
         get_raw_fastq
     output:
-        fastq1="trimmed/{sample}-{unit}.1.fastq.gz",
-        fastq2="trimmed/{sample}-{unit}.2.fastq.gz",
-        qc="trimmed/{sample}-{unit}.paired.qc.txt"
+        fastq1="results/trimmed/{sample}-{unit}.1.fastq.gz",
+        fastq2="results/trimmed/{sample}-{unit}.2.fastq.gz",
+        qc="results/trimmed/{sample}-{unit}.qc.txt"
```

Comment:
Divergence again. Please keep the `.paired.` infix of the qc file and add a `.single` infix to the one in the rule below.